doc/kernel/other/ring_buffers.rst

```diff
@@ -21,27 +22,66 @@ by its memory address.
-  that have been added to the ring buffer but not yet removed.
+* A **data buffer** of 32-bit words or bytes. The data buffer contains the data
+  items or raw bytes that have been added to the ring buffer but not yet
+  removed.
 
-* A **data buffer size**, measured in 32-bit words. This governs the maximum
-  amount of data (including metadata values) the ring buffer can hold.
+* A **data buffer size**, measured in 32-bit words or bytes. This governs the
+  maximum amount of data (including metadata values) the ring buffer can hold.
 
 A ring buffer must be initialized before it can be used. This sets its
 data buffer to empty.
 
+Data item mode
+==============
+
+Ring buffer instance works in data item mode if declared using
+:cpp:func:`SYS_RING_BUF_DECLARE_POW2()` or
+:cpp:func:`SYS_RING_BUF_DECLARE_SIZE()` and accessed using
+:cpp:func:`sys_ring_buf_put()` and :cpp:func:`sys_ring_buf_get()`.
+
 A ring buffer **data item** is an array of 32-bit words from 0 to 1020 bytes
-in length. When a data item is **enqueued** its contents are copied
-to the data buffer, along with its associated metadata values (which occupy
-one additional 32-bit word).
-If the ring buffer has insufficient space to hold the new data item
-the enqueue operation fails.
-
-A data items is **dequeued** from a ring buffer by removing the oldest
-enqueued item. The contents of the dequeued data item, as well as its
-two metadata values, are copied to areas supplied by the retriever.
-If the ring buffer is empty, or if the data array supplied by the retriever
-is not large enough to hold the data item's data, the dequeue operation fails.
+in length. When a data item is **enqueued** (:cpp:func:`sys_ring_buf_put()`)
+its contents are copied to the data buffer, along with its associated metadata
+values (which occupy one additional 32-bit word). If the ring buffer has
+insufficient space to hold the new data item the enqueue operation fails.
+
+A data items is **dequeued** (:cpp:func:`sys_ring_buf_get()`) from a ring buffer
+by removing the oldest enqueued item. The contents of the dequeued data item,
+as well as its two metadata values, are copied to areas supplied by the
+retriever. If the ring buffer is empty, or if the data array supplied by the
+retriever is not large enough to hold the data item's data, the dequeue
+operation fails.
+
+Byte mode
+=========
+
+Ring buffer instance works in byte mode if declared using
+:cpp:func:`SYS_RING_BUF_BYTES_DECLARE_SIZE()` and accessed using:
+:cpp:func:`sys_ring_buf_bytes_alloc()`, :cpp:func:`sys_ring_buf_bytes_put()`,
+:cpp:func:`sys_ring_buf_bytes_get()`, :cpp:func:`sys_ring_buf_bytes_free()`,
+:cpp:func:`sys_ring_buf_bytes_cpy_put()` and
+:cpp:func:`sys_ring_buf_bytes_cpy_get()`.
+
+Data can be copied into the ring buffer (see
+:cpp:func:`sys_ring_buf_bytes_cpy_put()`) or ring buffer memory can be used
+directly by the user. In the latter case, operation is splitted into three
```

Comment:
change to (note the blank line needed before a list):

    In the latter case, the operation is split into three stages:

    1. allocating the buffer (:cpp:func:`sys_ring_buf_bytes_alloc()`) when
       user requests the destination location where data can be written.
    #. writing the data by the user (e.g. buffer written by DMA).
    #. indicating the amount of data written to the provided buffer
       (:cpp:func:`sys_ring_buf_bytes_put()`). The amount
       can be less than or equal to the allocated amount.